src/bin/encode.rs

```diff
@@ -102,13 +102,35 @@ fn handle_path<'a>(options: &mut Options<'a>,
         progress!(options.quiet, "Compressing to memory");
     }
 
-    let source_len = std::fs::metadata(source_path)
-        .expect("Could not open source")
-        .len();
-
     progress!(options.quiet, "Parsing.");
-    let json = options.parser.parse_file(source_path)
-        .expect("Could not parse source");
+
+    handle_path_or_text(options, Some(source_path), None,
+                        dest_bin_path, dest_txt_path);
+}
+
+fn handle_path_or_text<'a>(options: &mut Options<'a>,
+    source_path: Option<&Path>,
```

Comment:
Four out of the five arguments look near exchangeable. Could you create a `struct` for all these arguments? Or make them part of `Options`?